Azure/azure-service-operator
# Add tests to ResourceGroup_Controller

**What this PR does / why we need it**:
This PR adds additional tests to the resourcegroup_controller_test.go file. 
This is necessary because, while using TDD to recreate the resourcegroup operator as an upskilling exercise, I discovered that you could get all of the current resource group controller tests to pass without ever creating or deleting anything in kubernetes or azure.
This PR adds in tests that prove that the resource group controller actually creates and deletes resource group instances. 

**Special notes for your reviewer**:
To test this:
- comment out lines 97-98, 108-130 in controllers/resourcegroup_controller.go
- run "make test" and observe that the tests in controllers/resourcegroup_controller_test.go still all pass
- now checkout the version of controllers/resourcegroup_controller_test.go contianed in this PR and see that the tests fail as expected with the code commented out. 
- uncomment the code to see the tests pass

**If applicable**:
- [ X] this PR contains tests
